src/itemStruct.ts

```diff
@@ -1,6 +1,59 @@
 import * as vscode from "vscode"
-import { getAutocompleteDump } from "./autocompleteDump"
-import { getApiDump, UNCREATABLE_TAGS } from "./dump"
+import { AutocompleteFunction, getAutocompleteDump } from "./autocompleteDump"
+import { ApiClass, ApiDump, getApiDump, UNCREATABLE_TAGS } from "./dump"
+
+const parameterClassFilter = (objectType: string, constraint: string) => {
+    return (klass: ApiClass): boolean => {
+        if (objectType === "Instance") {
+            if (constraint === "any") {
+                return true
+            } else if (constraint === "isScriptCreatable") {
+                const tags = klass.Tags
+                if (tags !== undefined) {
+                    for (const tag of tags) {
+                        if (UNCREATABLE_TAGS.has(tag)) {
+                            return false
+                        }
+                    }
+                }
+                return true
+            }
+        }
+        return false
+    }
+}
+
+const getFunctionParameters = (func: AutocompleteFunction, apiDump: ApiDump): [ string[], vscode.SnippetString ] => {
+    const insertText = new vscode.SnippetString(`${func.name}(`)
+
+    const params = []
+    for (const paramIndex in func.parameters) {
+        if (func.parameters[paramIndex] !== undefined) {
+            const param = func.parameters[paramIndex]
```

Comment:
This should be `for (const param of func.parameters)`

Reply:
I can't actually do this because I need to use the paramIndex later - unless you want me to use `for ... of` and use `Array.findIndex()` afterwards?

Reply:
You should be able to do something like `for (const [paramIndex, param] of Object.entries(func.parameters))`.